refactor: 상담 세션 통계 DTO 및 서비스 로직 수정

jawsbaek opened the pull request:
## 🔍️ 이 PR을 통해 해결하려는 문제가 무엇인가요?

- 상담 통계 기획 맞춰서 변경

## ✨ 이 PR에서 핵심적으로 변경된 사항은 무엇일까요?

- CounselSessionStatRes 클래스의 필드 변경
- 상담 세션 통계 조회 로직에서 약물 상담 수 및 상담사 수 추가
- 상담 세션 리포지토리에 상담사 수를 세는 메서드 추가
- MedicationCounselRepository에 생성일 범위로 카운트하는 메서드 추가

## 🔖 핵심 변경 사항 외에 추가적으로 변경된 부분이 있나요?

- 없음

## 🙏 Reviewer 분들이 이런 부분을 신경써서 봐 주시면 좋겠어요

> 개발 과정에서 다른 분들의 의견은 어떠한지 궁금했거나 크로스 체크가 필요하다고 느껴진 코드가 있다면 남겨주세요

-

## 🩺 이 PR에서 테스트 혹은 검증이 필요한 부분이 있을까요?

> 테스트가 필요한 항목이나 테스트 코드가 추가되었다면 함께 적어주세요

*

##### 📌 PR 진행 시 이러한 점들을 참고해 주세요

```
- Reviewer 분들은 코드 리뷰 시 좋은 코드의 방향을 제시하되, 코드 수정을 강제하지 말아 주세요.
- Reviewer 분들은 좋은 코드를 발견한 경우, 칭찬과 격려를 아끼지 말아 주세요.
- Review는 특수한 케이스가 아니면 Reviewer로 지정된 시점 기준으로 7일 이내에 진행해 주세요.
- Comment 작성 시 Prefix로 P1, P2, P3, P4, P5 를 적어 주시면 Assignee가 보다 명확하게 Comment에 대해 대응할 수 있어요
    - P1 : 꼭 반영해 주세요 (Request Changes) - 이슈가 발생하거나 취약점이 발견되는 케이스 등
    - P2 : 반영을 적극적으로 고려해 주시면 좋을 것 같아요 (Comment)
    - P3 : 이런 방법도 있을 것 같아요~ 등의 사소한 의견입니다 (Chore)
    - P4: 반영해도 좋고 넘어가도 좋습니다 (Approve)
    - P5: 그냥 사소한 의견입니다 (Approve+Chore)
```

---

jawsbaek committed:
refactor: 상담 세션 통계 DTO 및 서비스 로직 수정

- CounselSessionStatRes 클래스의 필드 변경
- 상담 세션 통계 조회 로직에서 약물 상담 수 및 상담사 수 추가
- 상담 세션 리포지토리에 상담사 수를 세는 메서드 추가
- MedicationCounselRepository에 생성일 범위로 카운트하는 메서드 추가

rappire commented on src/main/java/com/springboot/api/counselsession/service/CounselSessionService.java:
```diff
@@ -301,21 +301,30 @@ public List<LocalDate> getSessionDatesByYearAndMonth(int year, int month) {
-        Long counseleeCount = counselSessionRepository.countDistinctCounseleeForCurrentMonth();
-        long totalCaringMessageCount = counselSessionRepository.count();
-        double counselHours = calculateCounselHoursForThisMonth();
+        long counselHoursThisMonth = (long) calculateCounselHoursForThisMonth();
```
P3: calculateCounselHoursForThisMonth 이거 코드 봤는데, 이거 리턴이 double인 이유가 있나요?
long으로 바꿔도 문제없어보이네요

jawsbaek replied:
네넹

rappire reviewed (approved):
LGTM 좋습니다.

jawsbaek committed:
refactor: MedicationCounselRepository 메서드 이름 변경 및 상담 시간 계산 로직 수정

- MedicationCounselRepository에서 생성일 범위 카운트 메서드 이름을 countByCreatedDateBetween에서 countByCreatedDatetimeBetween으로 변경
- CounselSessionService에서 상담 시간 계산 로직을 long 타입으로 수정하고, 메서드 호출 방식 개선
